rMATS_pipeline/rmatspipeline/rmatspipeline.pyx

```diff
@@ -2926,20 +2941,27 @@ cdef size_t _load_job(str rmatsf, list vbams, list flags, const int& rl,
             print '         Please check %s' % (rmatsf)
 
         for i in range(len(bams)):
-            idx = vbams.index(bams[i])
+            idx = try_get_index(vbams, bams[i], &index_found)
+            if not index_found:
+            # It's ok to have data for bams besides vbams
+                continue
+
             novel_juncs[idx].clear()
             exons[idx].clear()
             juncs[idx].clear()
-            flags[idx] += 1
+            prep_counts_by_bam[idx] += 1
 
         # processing novel junctions
         for i in range(len(bams)):
             num = int(fp.readline())
-            idx = vbams.index(bams[i])
+            idx = try_get_index(vbams, bams[i], &index_found)
```

Comment:
I don't think 'idx = try_get_index(vbams, bams[i], &index_found)' here is necessary. Since this sanity check has already been performed in the first for loop.

Reply:
For the bams that are in the .rmats file but not in either sample try_get_index is needed in all the loops. The first loop initializes the data structures and when there is no index from try_get_index, then no initialization is needed for that loop iteration. The other loops are reading the data from the .rmats files and when there is no index from try_get_index, the lines in the .rmats file for those bams still need to be skipped over with fp.readline(), but the code for storing the data for those lines is skipped since there is no valid index